### DISCUSSION TITLE
DonutChartMetric::make

### DISCUSSION THREAD

**Discussion by Mr-Coquin** [moonshine-software/moonshine#1059] [Q&A] [answered]
Подскажите в FormPage страницы есть такая конструкция
 public function fields(): array
    {
return [

Modal::make(
                            title: 'Confirm',
                            content: 'Detail information',
                            outer: ActionButton::make('Show modal', '#'),
                            components: PageComponents::make([
                             DonutChartMetric::make('Subscribers')
                                                            ->values([
                                                               "sum1"=>(int) $this->getResource()->getItem()->sum1,
                                                                "sum2"=>(int) $this->getResource()->getItem()->sum2
                                                            ])
                                                            
Можно ли для круговова графика как то более правильно взять данные и передать не только целые числа ?

**Comment by Mr-Coquin** [comment]
Меня смущает как я взял данные (int) $this->getResource()->getItem()->sum1

**Reply by DissNik** [reply, marked as answer]
https://moonshine-laravel.com/docs/resource/components/components-metric_donut_chart#decimals

Можно использовать не только целые числа

**Reply by Mr-Coquin** [reply]
> https://moonshine-laravel.com/docs/resource/components/components-metric_donut_chart#decimals
> 
> Можно использовать не только целые числа

да, успел сам найти . Надо было 
                                                            ->values([
                                                                "sum1"=>(float)$this->getResource()->getItem()->sum1,
                                                                "sum2"=>(float)$this->getResource()->getItem()->sum2

                                                            ])
                                                            
                                                            Не подскажите можно ли как то красивей взять данные не через такую связку sum2"=>(float)$this->getResource()->getItem()->sum2 ???

**Comment by lee-to** [comment]
Кстати вопросы я думаю лучше в чате задавать а тут больше для фича реквест

**Reply by Mr-Coquin** [reply]
Там часто остается вопрос не замеченным из-за кол-ва сообщений иногда. Я бываю пол дня не захожу и вуаля 150 новых сообщений :)

**Reply by Mr-Coquin** [reply]
Кстати надо внедрять бы платную систему поддержки в виде тикетов :) я бы первый побежал бы подписываться туда еще кроме как спонсорства

**Reply by lee-to** [reply]
Я пока тяжело себе представляю как это будет выглядеть, многое в тикетах то что придется делать платно мы не сможем добавить в коробку так как коробка должна быть как можно тоньше и только со всем необходимым, а писать пакеты по заказу это будет трудоемко. в итоге такая платная система будет тем же самым что есть сейчас в чате и здесь в дискуссиях, но только платно)

**Reply by lee-to** [reply]
Да и спасибо вам за поддержку!

**Reply by Mr-Coquin** [reply]
> Я пока тяжело себе представляю как это будет выглядеть, многое в тикетах то что придется делать платно мы не сможем добавить в коробку так как коробка должна быть как можно тоньше и только со всем необходимым, а писать пакеты по заказу это будет трудоемко. в итоге такая платная система будет тем же самым что есть сейчас в чате и здесь в дискуссиях, но только платно)

Ну я не говорю что это будет панацея. Но как дополнительный доход для проекта может быть. Я на примере некоторых систем знаю, что можно бесплатно всегда написать но там ответ будет в течение 1-3 суток + он может содержать краткий ответ или просто ссылку. Но когда есть возможность например написать по подписке из личного кабинета в зависимости от плана к примеру в сутки 5-10 вопросов разработчикам: "Как мне сделать множественный выбор в Select" -> в чате 100% ответ будет ссылка на доку и все. Но в Moonshine приходят иногда люди далекие о программирования и слабые или ленивые в понимание документации и получая ответ пример запроса с ->multiple()  уровень удовлетворения будет колоссальный.  По себе знаю, ищу иногда в документации что то (я точно помню что это видел и потом нахожу через часа 2-3), AI выдает такую дичь а в чате никто не отвечает и раздражённость просто через край прёт :) . А так бы написал платный запрос и взялся бы за другую задачу. 

Я к чему - решать вам как разработчикам, но думаю нам потратиться как фанатам moonshine пару баксами для удобства и поддержки проекта не составит труда.

**Reply by lee-to** [reply]
> > Я пока тяжело себе представляю как это будет выглядеть, многое в тикетах то что придется делать платно мы не сможем добавить в коробку так как коробка должна быть как можно тоньше и только со всем необходимым, а писать пакеты по заказу это будет трудоемко. в итоге такая платная система будет тем же самым что есть сейчас в чате и здесь в дискуссиях, но только платно)
> 
> 
> 
> Ну я не говорю что это будет панацея. Но как дополнительный доход для проекта может быть. Я на примере некоторых систем знаю, что можно бесплатно всегда написать но там ответ будет в течение 1-3 суток + он может содержать краткий ответ или просто ссылку. Но когда есть возможность например написать по подписке из личного кабинета в зависимости от плана к примеру в сутки 5-10 вопросов разработчикам: "Как мне сделать множественный выбор в Select" -> в чате 100% ответ будет ссылка на доку и все. Но в Moonshine приходят иногда люди далекие о программирования и слабые или ленивые в понимание документации и получая ответ пример запроса с ->multiple()  уровень удовлетворения будет колоссальный.  По себе знаю, ищу иногда в документации что то (я точно помню что это видел и потом нахожу через часа 2-3), AI выдает такую дичь а в чате никто не отвечает и раздражённость просто через край прёт :) . А так бы написал платный запрос и взялся бы за другую задачу. 
> 
> 
> 
> Я к чему - решать вам как разработчикам, но думаю нам потратиться как фанатам moonshine пару баксами для удобства и поддержки проекта не составит труда.   
> 
> 

Хорошая идея, обязательно подумаем после выхода 3.0
